stdlib/universe/yield.go

```diff
@@ -17,7 +20,21 @@ type YieldOpSpec struct {
 func init() {
 	yieldSignature := semantic.MustLookupBuiltinType("universe", "yield")
 
-	flux.RegisterPackageValue("universe", YieldKind, flux.MustValue(flux.FunctionValueWithSideEffect(YieldKind, createYieldOpSpec, yieldSignature)))
+	flux.RegisterPackageValue("universe", YieldKind, values.NewFunction("yield", yieldSignature, func(ctx context.Context, args values.Object) (values.Value, error) {
+		in, ok := args.Get("tables")
+		if !ok {
+			return nil, errors.Newf(codes.Invalid, "missing required keyword argument %q", "tables")
+		}
+
+		if _, ok := in.(*flux.TableObject); ok {
+			fn, err := flux.FunctionValueWithSideEffect("yield", createYieldOpSpec, yieldSignature)
+			if err != nil {
+				return nil, err
+			}
+			return fn.Function().Call(ctx, args)
+		}
+		return in, nil
+	}, true))
```

Comment:
So this says that instead of working only on tables, `yield` is essentially the identity function? So now you can yield a value of any type?

Reply:
This is correct. I can try to get a different way for this to work if we don't want this, but I figured we wanted to change `yield` at some point anyway to do this and I needed a side effect and it isn't possible to inject side effect functions anymore.